src/main/redux/sagas/persist.ts

```diff
@@ -8,12 +8,12 @@
 import * as debug_ from "debug";
 import { selectTyped } from "readium-desktop/common/redux/sagas/typed-saga";
 import { ConfigRepository } from "readium-desktop/main/db/repository/config";
-import { CONFIGREPOSITORY_REDUX_PERSISTENCE, diMainGet } from "readium-desktop/main/di";
+import { CONFIGREPOSITORY_REDUX_PERSISTENCE, diMainGet, setReduxPersistState } from "readium-desktop/main/di";
 import { winActions } from "readium-desktop/main/redux/actions";
 import { RootState } from "readium-desktop/main/redux/states";
 import { call, debounce  } from "redux-saga/effects";
 
-const DEBOUNCE_TIME = 1000;
+const DEBOUNCE_TIME = 30000;
```

Comment:
On first impressions, this seems like a very long time (for example, AudioBook playback notifications are throttled at 1s intervals, so there would be many missed reading location updates).
However, pass the first impressions, I am wondering whether the state is immediately saved to persistent storage when closing the app? To me, this is the main scenario that needs to be covered: user reads / listens to (audio)book, quickly close the app => we must make sure all current settings are preserved, especially the current reading location which accounts for the highest notification / event traffic / relatively high frequency ... which is why the database "compact" MUST work reliably (currently does not seem to :(

Reply:
indeed the state is saved to persistent storage on the closing process

Reply:
So I guess this leaves the possibility of a sudden application crash, in which case the worst case scenario is that 30s worth of state changes can would be missed.